pkg/discovery/processor/cluster_processor.go

```diff
@@ -229,6 +229,14 @@ func (p *ClusterProcessor) DiscoverCluster() (*repository.ClusterSummary, error)
 	// Discover Turbo Policies
 	NewTurboPolicyProcessor(p.clusterInfoScraper, kubeCluster).ProcessTurboPolicies()
 
+	// Discover GitOps configuration overrides
+	NewGitOpsConfigProcessor(p.clusterInfoScraper, kubeCluster).ProcessGitOpsConfigs()
+
+	// Update the GitOps configuration override cache
+	if clusterScraper, ok := p.clusterInfoScraper.(*cluster.ClusterScraper); ok {
+		clusterScraper.UpdateGitOpsConfigCache(kubeCluster.GitOpsConfigurations)
```

Comment:
This is an odd usage. We are sort of duplicating the cache data in memory. Effectively storing the copy in `kubeCluster` and then in `clusterScraper.GitopsConfigCache`. We need only one copy. I would simply call `GetAllGitOpsConfigurations()` here and pass the data on. We don't need the copy in `kubeCluster`. Or do we, I might be missing something.. :-)

Reply:
Updated to call `UpdateGitOpsConfigCache` which in turn calls `GetAllGitOpsConfigurations`.